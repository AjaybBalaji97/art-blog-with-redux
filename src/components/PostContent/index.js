import React, { Component } from 'react';
import './PostContent.css';
class PostContent extends Component {
    render() {
      const product = this.props.product;
  
      return (
        <div className="main" >
                <div className="backgroundImage" style={{ backgroundImage: `url("./Images/${product.image}")` }} />

                <div className="imageInfo">

                    <span>{product.date}</span><span>{product.readingTime}</span>
                </div>
                <div className="imageDesc">
                    <span >{product.title}</span>
                    <span className="imageContent">{product.description}</span>

                </div>
                <hr />
        </div>
      );
    }
  }
  export default PostContent;