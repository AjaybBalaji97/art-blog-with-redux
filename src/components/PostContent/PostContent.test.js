import React from 'react';
import renderer from 'react-test-renderer';
import PostContent from '.';
import {shallow} from 'enzyme';


describe('PostContent',()=>{
    it('should render header',()=>{
        const products={
            "date": "2nd Januray, 2018",
            "readingTime": "2 mins",
            "title": "The future of abstract art and the culture ...",
            "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
            "claps": 10,
            "liked": false,
            "image": "abstract.png"
        }
        const tree=renderer.create(<PostContent product={products} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});