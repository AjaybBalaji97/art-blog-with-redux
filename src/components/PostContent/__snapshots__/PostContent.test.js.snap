// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostContent should render header 1`] = `
<div
  className="main"
>
  <div
    className="backgroundImage"
    style={
      Object {
        "backgroundImage": "url(\\"./Images/abstract.png\\")",
      }
    }
  />
  <div
    className="imageInfo"
  >
    <span>
      2nd Januray, 2018
    </span>
    <span>
      2 mins
    </span>
  </div>
  <div
    className="imageDesc"
  >
    <span>
      The future of abstract art and the culture ...
    </span>
    <span
      className="imageContent"
    >
      Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
    </span>
  </div>
  <hr />
</div>
`;
